crates/hir_ty/src/diagnostics/expr.rs

```diff
@@ -155,14 +155,17 @@ impl<'a, 'b> ExprValidator<'a, 'b> {
 
         // FIXME: Due to shortcomings in the current type system implementation, only emit this
         // diagnostic if there are no type mismatches in the containing function.
-        if self.infer.type_mismatches.iter().next().is_some() {
-            return Some(());
-        }
+        // if self.infer.type_mismatches.iter().next().is_some() {
```

Comment:
I'm not sure if removing this check was a good decision

Reply:
Hm, I guess I am wondering a bigger thing: whey do we want to piggy-back this on the `validate_call` function? 

I think this diagnostic should be applicable anywhere where there's a type mismatch, and we should keep `validate_call` intact, to only process mismatched arg counts.

Reply:
I added support for `LetStmt`, yet I still can't find any way to completely remove this diagnostic from validate_call